### PR TITLE
Implements tail() for DataFrame & Series

### PR DESCRIPTION
**Pull request by itholic**
Implements pandas.DataFrame.tail() (https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.tail.html#pandas.DataFrame.tail) and pandas.Series.tail(https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.tail.html#pandas.Series.tail)

For DataFrame

```python
>>> df = ks.DataFrame({'animal':['alligator', 'bee', 'falcon', 'lion',
...                    'monkey', 'parrot', 'shark', 'whale', 'zebra']})
>>> df
      animal
0  alligator
1        bee
2     falcon
3       lion
4     monkey
5     parrot
6      shark
7      whale
8      zebra

>>> df.tail(3)
  animal
8  zebra
7  whale
6  shark
```

For Series

```python
>>> df = ks.DataFrame({'animal':['alligator', 'bee', 'falcon', 'lion']})
>>> df.animal.tail(2)
3      lion
2    falcon
Name: animal, dtype: object
```

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -4229,6 +4229,63 @@ def head(self, n=5):
...
+
+        Viewing the last `n` lines (three in this case)
+
+        >>> df.tail(3)
+          animal
+        8  zebra
+        7  whale
+        6  shark
+        """
+        tmp_col = '__order__'
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: I would name it `__natural_order__`.

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -4229,6 +4229,63 @@ def head(self, n=5):
...
+        sdf = self._sdf.withColumn(tmp_col, F.monotonically_increasing_id())
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: Actually I think we don't need to add this as a column. Can we `sdf.orderBy(F.monotonically_increasing_id())` directly?

**Comment by HyukjinKwon** [comment]
Looks okay.

File: databricks/koalas/frame.py (hunk 2 above)
Comment (itholic) [reply databricks/koalas/frame.py]: it works 👍 i just changed the logic like this.

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -4229,6 +4229,62 @@ def head(self, n=5):
...
         return DataFrame(self._internal.copy(sdf=self._sdf.limit(n)))
 
+    def tail(self, n=5):
+        """
+        Return the last `n` rows.
+
+        This function returns the last `n` rows for the object based
+        on position. It is useful for quickly verifying data,
+        for example, after sorting or appending rows.
+
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: FYI, I plan to propose a new API `tail` in Spark itself. If we manage to add it, we could efficiently do that.

Comment (HyukjinKwon) [reply databricks/koalas/frame.py]: Can we hold on for a while? We could leave a note about that if we add it in Spark.

Comment (itholic) [reply databricks/koalas/frame.py]: @HyukjinKwon ah, that sounds great!! 👍 
Okay, then just let me know for fix this after `tail` will be added on spark,

Comment (HyukjinKwon) [reply databricks/koalas/frame.py]: PR submitted at https://github.com/apache/spark/pull/26809. Let's see how it goes.

Comment (itholic) [reply databricks/koalas/frame.py]: @HyukjinKwon awesome! thanks for reminding me 👍

**Comment by HyukjinKwon** [comment]
Now, Apache Spark 3.0 will have an API DataFrame.tail. Let's revisit this when Spark 3.0 is out.

**Pull request closed without merging**